### PR TITLE
Settings Routes

### PR DESCRIPTION
**Pull request by tommoor**
Recreates the current settings under new routes with a new settings sidebar and a tweaked header to show the org logo. This is a stepping stone towards a fuller settings area mocked here:
https://share.goabstract.com/872e85f7-6afb-4de3-acf9-c6396fc31a8e

image

---

image

---

image

### CODE REVIEW

File: app/components/Layout/Layout.js [hunk 1]
```diff
@@ -84,7 +85,12 @@ class Layout extends React.Component {
         {this.props.notifications}
 
         <Flex auto>
-          {showSidebar && <Sidebar />}
+          {showSidebar && (
+            <Switch>
+              <Route path="/settings" component={SettingsSidebar} />
```
Comment (tommoor) [inline app/components/Layout/Layout.js]: All settings routes have to be under `/settings` for this to work, seems reasonable though.

File: app/components/Sidebar/components/Collections.js [hunk 2]
```diff
@@ -261,15 +262,6 @@ const StyledDropToImport = styled(DropToImport)`
   }
 `;
 
-const Header = styled(Flex)`
```
Comment (tommoor) [inline app/components/Sidebar/components/Collections.js]: Moved out to it's own file

File: app/components/SlackAuthLink/SlackAuthLink.js [hunk 3]
```diff
@@ -1,22 +0,0 @@
...
-import AuthStore from 'stores/AuthStore';
-
-type Props = {
-  children: React$Element<*>,
-  auth: AuthStore,
-  scopes?: string[],
-  redirectUri?: string,
-};
-
-function SlackAuthLink({ auth, children, scopes, redirectUri }: Props) {
```
Comment (tommoor) [inline app/components/SlackAuthLink/SlackAuthLink.js]: Became `SlackButton`

File: app/scenes/Dashboard/Dashboard.js [hunk 4]
```diff
@@ -2,26 +2,15 @@
...
 
 import DocumentsStore from 'stores/DocumentsStore';
 import Flex from 'shared/components/Flex';
 import DocumentList from 'components/DocumentList';
 import PageTitle from 'components/PageTitle';
+import Subheading from 'components/Subheading';
 import CenteredContent from 'components/CenteredContent';
 import { ListPlaceholder } from 'components/LoadingPlaceholder';
 
-const Subheading = styled.h3`
```
Comment (tommoor) [inline app/scenes/Dashboard/Dashboard.js]: Moved to `Subheading` so it's more easily shareable

File: app/scenes/Settings/components/ApiToken.js [hunk 5]
```diff
@@ -32,21 +31,14 @@ class ApiKeyRow extends React.Component {
         <td>
           <code>{secret}</code>
         </td>
-        <td>
-          <Action role="button" onClick={this.onClick} disabled={disabled}>
-            Action
```
Comment (tommoor) [inline app/scenes/Settings/components/ApiToken.js]: This literally said the word "Action", lol.

File: app/stores/SettingsStore.js [hunk 6]
```diff
@@ -4,11 +4,10 @@ import invariant from 'invariant';
 import { client } from 'utils/ApiClient';
 import type { ApiKey } from 'types';
 
-class SearchStore {
+class SettingsStore {
```
Comment (tommoor) [inline app/stores/SettingsStore.js]: Note, this store moved to the `/stores/` repo out of Settings scene.

File: app/components/Sidebar/Settings.js [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+    this.scrollable = ref;
+  };
+
+  returnToDashboard = () => {
+    this.props.history.push('/');
+  };
+
+  render() {
+    const { auth } = this.props;
+    const { team } = auth;
```
Comment (jorilallo) [inline app/components/Sidebar/Settings.js]: `const { team } = this.props.auth;`

**Pull request merged**